components/developer/eclipse/Makefile

```diff
@@ -0,0 +1,57 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2017 Aurelien Larcher
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=         eclipse
+COMPONENT_VERSION=      4.6.3
+COMPONENT_SUMMARY=      Eclipse IDE for Java, C/C++, Python, JavaScript and PHP
+COMPONENT_PROJECT_URL=  https://eclipse.org/
+COMPONENT_FMRI=         developer/eclipse
+COMPONENT_CLASSIFICATION=Development/Editors
+COMPONENT_SRC=          $(COMPONENT_NAME)
+COMPONENT_ARCHIVE=      $(COMPONENT_NAME)-$(COMPONENT_VERSION).zip
+COMPONENT_ARCHIVE_DATE= 201703010400
+COMPONENT_ARCHIVE_URL= \
+  http://mirrors.nic.cz/eclipse/eclipse/downloads/drops4/R-$(COMPONENT_VERSION)-$(COMPONENT_ARCHIVE_DATE)/eclipse-platform-$(COMPONENT_VERSION)-solaris-gtk-x86_32.zip
```

Comment:
I think redistributing binaries is good for now, I just patcheed the .ini file to circumvent the GTK issue. Distributions like Arch Linux distribute the binaries.
We may need to update other components (ant?) before we can build it.